src/csharp/Microsoft.Spark/Sql/Functions.cs

```diff
@@ -31,13 +32,13 @@ public static Column Column(string columnName)
             return ApplyFunction("col", columnName);
         }
 
-        public static Column ApplyTonicUdf(Column col, string replacement, string statistics, int baseSeed)
+        public static Column ApplyTonicUdf(Column col, string replacement, string statistics, int baseSeed, JavaUnsignedLongAccumulator processedRowCounter)
         {
             return new Column(
                 (JvmObjectReference)Jvm.CallStaticJavaMethod(
                     "com.tonic.udfs.GeneratorCaller",
                     "create",
-                    new object[] {col, replacement, statistics, baseSeed}));
+                    new object[] {col, replacement, statistics, baseSeed, processedRowCounter}));
```

Comment:
Is this PR supposed to go out before [the java one](https://github.com/TonicAI/allos/pull/7662/files#diff-6915291bd86d60ac4d077c6cdf41b4f161e7e9e673cf7eae773ba1d698028de4R38) or vice versa?

Reply:
We copy a built `Microsoft.Spark.dll` as well as a patch for the `Microsoft.Spark.Worker` library in the allos repository. So all of the code changes in this PR are already included in https://github.com/TonicAI/allos/pull/7662

Ideally this should be approved and merged before https://github.com/TonicAI/allos/pull/7662 is merged in.

Reply:
I see - so the spark worker code in here is already technically in there (and any nitpick style changes you make here need to be recompiled and uploaded there 😅 ).

Alright then. Thanks for the explanation!